core/src/main/java/io/parsingdata/metal/Util.java

```diff
@@ -111,6 +118,33 @@ public Optional<Value> eval(final Value value, final ParseState parseState, fina
         };
     }
 
+    public static ValueExpression digest(final String algorithm, final ValueExpression target) {
+        return new UnaryValueExpression(target) {
+            @Override
+            public Optional<Value> eval(final Value value, final ParseState parseState, final Encoding encoding) {
+                try {
+                    final Slice slice = value.slice();
+                    final BigInteger bufferSize = valueOf(512);
+                    final MessageDigest digester = getInstance(algorithm);
+
+                    for (BigInteger i = ZERO; i.compareTo(slice.length) < 0; i = i.add(bufferSize)) {
+                        final BigInteger length = bufferSize.min(slice.length.subtract(i));
+                        final Optional<Slice> subslice = createFromSource(slice.source, slice.offset.add(i), length);
+                        if (subslice.isEmpty()) {
+                            return Optional.empty();
+                        }
+                        digester.update(subslice.get().getData());
```

Comment:
Can we also do without this `get()`?

Also, I think the windowing part is very interesting and other implementations may benefit from it was well. We may consider to extract the windowing part to a separate value expression so we can reuse it for other implementations.